packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -982,6 +990,13 @@ Hardhat Network's forking functionality only works with blocks from at least spu
       mixHashGenerator: this._mixHashGenerator.clone(),
     };
 
+    const rethnetSnapshot = await this._rethnetState.makeSnapshot();
```

Comment:
Shouldn't there be a matching `this._rethnetState.removeSnapshot` call?

Reply:
The problem is that I wouldn't know when to actually remove the snapshot. For now, all snapshots keep existing.

I made a tracking issue for this effort: https://github.com/NomicFoundation/rethnet/issues/33